pkgs/development/compilers/rust/rustc.nix

```diff
@@ -75,6 +75,7 @@ in stdenv.mkDerivation rec {
     "--release-channel=stable"
     "--set=build.rustc=${rustc}/bin/rustc"
     "--set=build.cargo=${cargo}/bin/cargo"
+    "--tools=rustc"
```

Comment:
This is sweet.

Reply:
This change is causing problems with rust-analyzer, because of missing `rust-analyzer-proc-macro-srv` which can be usually found in `libexec` directory.

```
proc macro `Derivative` not expanded: cannot find proc-macro server in sysroot `/nix/store/yx6y8yrymng2pdfhpcrx26xb6sb21300-rustc-1.71.0`
```

On `master`:
```
$ nix-build -A rustc
/nix/store/1qj28hnb372f7fld7zlh5cr1bq5s8h50-rustc-1.70.0

$ ls -la /nix/store/1qj28hnb372f7fld7zlh5cr1bq5s8h50-rustc-1.70.0
total 186533
dr-xr-xr-x      6 root root        6 Jan  1  1970 .
drwxrwxr-t 194219 root nixbld 538180 Jul 24 14:31 ..
dr-xr-xr-x      2 root root        7 Jan  1  1970 bin
dr-xr-xr-x      3 root root        6 Jan  1  1970 lib
dr-xr-xr-x      2 root root        3 Jan  1  1970 libexec
dr-xr-xr-x      2 root root        3 Jan  1  1970 nix-support

$ ls -la /nix/store/1qj28hnb372f7fld7zlh5cr1bq5s8h50-rustc-1.70.0/libexec
total 1146
dr-xr-xr-x 2 root root       3 Jan  1  1970 .
dr-xr-xr-x 6 root root       6 Jan  1  1970 ..
-r-xr-xr-x 1 root root 2255528 Jan  1  1970 rust-analyzer-proc-macro-srv
```
With this patch:
```
$ nix-build -A rustc
/nix/store/yx6y8yrymng2pdfhpcrx26xb6sb21300-rustc-1.71.0

$ ls -la /nix/store/yx6y8yrymng2pdfhpcrx26xb6sb21300-rustc-1.71.0
total 186533
dr-xr-xr-x      5 root root        5 Jan  1  1970 .
drwxrwxr-t 194219 root nixbld 538180 Jul 24 14:31 ..
dr-xr-xr-x      2 root root        7 Jan  1  1970 bin
dr-xr-xr-x      3 root root        6 Jan  1  1970 lib
dr-xr-xr-x      2 root root        3 Jan  1  1970 nix-support
```
Maybe `rust-analyzer` needs to be wrapped differently ? cc @figsoda

Reply:
Ahh, I forgot the check `libexec`

fix in #245225, thanks for pointing this out